File: ios/RlyNetworkMobileSdk.m
#import <React/RCTBridgeModule.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RlyNetworkMobileSdk, NSObject)

RCT_EXTERN_METHOD(hello:
    (RCTPromiseResolveBlock) resolve
    rejecter: (RCTPromiseRejectBlock) reject
)

RCT_EXTERN_METHOD(getBundleId:
    (RCTPromiseResolveBlock) resolve
    rejecter: (RCTPromiseRejectBlock) reject
)

RCT_EXTERN_METHOD(getWalletMnemonic:
    (RCTPromiseResolveBlock) resolve
    rejecter: (RCTPromiseRejectBlock) reject
)

RCT_EXTERN_METHOD(saveWalletMnemonic:
    (NSString *) mnemonic
    resolver: (RCTPromiseResolveBlock) resolve
    rejecter: (RCTPromiseRejectBlock) reject
)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/RlyNetworkMobileSdk.swift
import Foundation
@objc(RlyNetworkMobileSdk)
class RlyNetworkMobileSdk: NSObject {
  var walletKey: String?  = nil

  @objc
  public func hello(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) -> Void {
    resolve("Hello World")
  }

  @objc
  public func getBundleId(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) -> Void {
    resolve(Bundle.main.bundleIdentifier!)
  }

  @objc
  public func getWalletMnemonic(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) -> Void {
    resolve(walletKey)
  }

  @objc
  public func saveWalletMnemonic(
    _ mnemonic: String,
    resolver resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) -> Void {
    walletKey = mnemonic
    resolve(true)
  }

}
